lib/chef/run_lock.rb

```diff
@@ -87,6 +88,9 @@ def acquire
     # Either acquire() or test() methods should be called in order to
     # get the ownership of run_lock.
     def test
+      if Pathname.new(runlock_file).relative?
+        @runlock_file = File.absolute_path(runlock_file, working_directory)
```

Comment:
That was my first option, but later I realised that doesn't work because the working directory of chef-client is /.

File.expand_path(runlock_file) returns "/foo/local-mode-cache/cache/chef-client-running.pid" if you run: ```chef-client --local --config foo/bar```.